File: apis/src/main/java/com/beat/apis/performance/application/dto/create/StaffResponse.java
package com.beat.apis.performance.application.dto.create;

import com.beat.global.support.jackson.CdnImageUrl;

public record StaffResponse(
	Long staffId,
	String staffName,
	String staffRole,
	@CdnImageUrl String staffPhoto
) {
	public static StaffResponse of(
		Long staffId,
		String staffName,
		String staffRole,
		String staffPhoto
	) {
		return new StaffResponse(
			staffId,
			staffName,
			staffRole,
			staffPhoto
		);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/StaffResponse.kt
package com.beat.apis.performance.api.response

import com.beat.application.frontoffice.performance.StaffResult
import com.beat.apis.web.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class StaffResponse private constructor(
    val staffId: Long?,
    val staffName: String?,
    val staffRole: String?,
    @field:CdnImageUrl val staffPhoto: String?,
) {
    companion object {
        fun from(result: StaffResult): StaffResponse = StaffResponse(result.id, result.name, result.role, result.photo)
    }
}
